modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DataNodesManagerTest.java

```diff
@@ -478,6 +496,75 @@ public void nodeRejoinWithDifferentAttributesWithAutoAdjustAlteration() throws I
         assertEquals(C_DIFFERENT_ATTRS.userAttributes().get("region"), c.userAttributes().get("region"));
     }
 
+    @Test
+    public void testHistoryCompaction() throws InterruptedException {
+        alterZone(ZONE_NAME_1, 0, 0, null);
+
+        removeNodes(Set.of(C));
```

Comment:
I do not understand, why do you remove C if C is not in the data nodes?

Reply:
removed